tests/installation/opensuse_welcome.pm

```diff
@@ -23,16 +23,14 @@ sub run {
     # Untick box - (Retries may be needed: poo#56024)
     for my $retry (1 .. 5) {
         assert_and_click_until_screen_change("opensuse-welcome-show-on-boot", 5, 5);
-        last unless check_screen("opensuse-welcome-show-on-boot");
+        # Moving the already causes screen changes - let any changes
+        # settle down before checking the screen contents.
+        wait_still_screen 2;
+        last if check_screen("opensuse-welcome-show-on-boot-unselected");
         die "Unable to untick 'Show on next startup'" if $retry == 5;
     }
 
-    # Close welcome screen - (Retries may be needed: poo#56024)
-    for my $retry (1 .. 5) {
-        assert_and_click_until_screen_change("opensuse-welcome-close-btn", 5, 5);
-        last unless check_screen("opensuse-welcome");
-        die "Unable to close openSUSE Welcome screen" if $retry == 5;
-    }
+    send_key_until_needlematch("generic-desktop", "alt-f4");
```

Comment:
That will be very tricky - as in first_boot, we have 'opensuse-welcome' as part of the detection screen - so it is likely part of some needles matching also 'generic-desktop'

Reply:
See #8386 
Removing the "generic-desktop" needles showing the dialog will also trim down the number of candidates for any later "generic-desktop" matches.

Reply:
I have changed this to only break on "dialog no longer visible". As the "unless check_screen" matches immediately while the dialog is still visible, i have nested the send key in a wait_screen_change.